src/Codegen/InputGenerator.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spawnia\Sailor\Codegen;
+
+use GraphQL\Type\Definition\EnumType;
+use GraphQL\Type\Definition\InputObjectType;
+use GraphQL\Type\Definition\ScalarType;
+use GraphQL\Type\Definition\Type;
+use GraphQL\Type\Schema;
+use Nette\PhpGenerator\ClassType;
+use Nette\PhpGenerator\PhpNamespace;
+use Spawnia\Sailor\EndpointConfig;
+
+class InputGenerator
+{
+    protected Schema $schema;
+
+    protected EndpointConfig $endpointConfig;
+
+    public function __construct(Schema $schema, EndpointConfig $endpointConfig)
+    {
+        $this->schema = $schema;
+        $this->endpointConfig = $endpointConfig;
+    }
+
+    /**
+     * @return iterable<ClassType>
+     */
+    public function generate(): iterable
+    {
+        foreach ($this->schema->getTypeMap() as $type) {
+            if (! $type instanceof InputObjectType) {
+                continue;
+            }
+
+            $class = new ClassType(
+                $type->name,
+                new PhpNamespace(static::inputNamespace($this->endpointConfig))
+            );
+
+            foreach ($type->getFields() as $name => $field) {
+                $property = $class->addProperty($name);
+
+                $fieldType = $field->getType();
+                $namedType = Type::getNamedType($fieldType);
+
+                if ($namedType instanceof ScalarType) {
+                    $typeReference = PhpType::forScalar($namedType);
+                } elseif ($namedType instanceof EnumType) {
+                    $typeReference = PhpType::forEnum($namedType);
+                } elseif ($namedType instanceof InputObjectType) {
+                    $typeReference = '\\'.static::className($namedType, $this->endpointConfig);
+                } else {
+                    // @phpstan-ignore-next-line
+                    throw new \Exception('Unsupported type '.get_class($namedType).' found.');
+                }
+
+                $property->setComment('@var '.PhpType::phpDoc($fieldType, $typeReference));
+            }
+
+            yield $class;
+        }
+    }
+
+    public static function className(InputObjectType $type, EndpointConfig $endpointConfig): string
+    {
+        return self::inputNamespace($endpointConfig).'\\'.$type->name;
+    }
+
+    protected static function inputNamespace(EndpointConfig $endpointConfig): string
+    {
+        return $endpointConfig->namespace().'\\Inputs';
```

Comment:
```suggestion
        return $endpointConfig->namespace().'\\Input';
```

I usually see singular form in NSs

Reply:
Depends, namespaces that are expected to have a flat list of monomorphic classes are usually plural in my experience. If there is more diversity among the contained classes, I would go for singular.

For example:
- `GraphQL\Type` holds all possible types, but also a lot of other stuff
- `GraphQL\Validator\Rules` exclusively holds classes that extend `ValidationRule`

I think this case falls into the second category. I will however change the name of the method, which is currently inconsistent with the namespace.